libpod/in_memory_state.go

```diff
@@ -245,6 +327,16 @@ func (s *InMemoryState) ContainerInUse(ctr *Container) ([]string, error) {
 		return nil, ErrCtrRemoved
 	}
 
+	// If the container does not exist, return error
+	if _, ok := s.containers[ctr.ID()]; !ok {
+		ctr.valid = false
+		return nil, errors.Wrapf(ErrNoSuchCtr, "container with ID %s not found in state", ctr.ID())
+	}
+
+	if err := s.checkNSMatch(ctr.ID(), ctr.Namespace()); err != nil {
+		return nil, err
```

Comment:
Should you set ctr.valid=false?

Reply:
It's still a valid container, just not in this namespace... I think leaving valid as-is is probably fine for this case.